File: order/src/main/java/com/yas/order/model/OrderAddress.java
package com.yas.order.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "order_address")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class OrderAddress {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String contactName;
    private String phone;
    private String addressLine1;
    private String addressLine2;
    private String city;
    private String zipCode;
    private Long districtId;
    private String districtName;
    private Long stateOrProvinceId;
    private String stateOrProvinceName;
    private Long countryId;
    private String countryName;
}

File: backoffice/modules/order/models/OrderAddress.ts
export type OrderAddress = {
  id?: number;
  contactName: string;
  phone: string;
  addressLine1: string;
  addressLine2?: string;
  city?: string;
  zipCode?: string;
  districtId: number;
  districtName?: string;
  stateOrProvinceId: number;
  stateOrProvinceName?: string;
  countryId: number;
  countryName?: string;
  isActive?: boolean;
};
